src/coreclr/nativeaot/Runtime/Full/CMakeLists.txt

```
@@ -6,7 +6,13 @@ project(Runtime)
 # Include auto-generated files on include path
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-add_definitions(-DFEATURE_RX_THUNKS)
+if (NOT CLR_CMAKE_TARGET_APPLE OR CLR_CMAKE_TARGET_OSX)
+  add_definitions(-DFEATURE_RX_THUNKS)
+else()
+  list(APPEND RUNTIME_SOURCES_ARCH_ASM
+    ${ARCH_SOURCES_DIR}/ThunkPoolThunks.${ASM_SUFFIX}
+  )
+endif()
```

I'd flip the condition, I find it easier to read that way:
```suggestion
if (CLR_CMAKE_TARGET_APPLE AND NOT CLR_CMAKE_TARGET_OSX)
  list(APPEND RUNTIME_SOURCES_ARCH_ASM
    ${ARCH_SOURCES_DIR}/ThunkPoolThunks.${ASM_SUFFIX}
  )
else()
  add_definitions(-DFEATURE_RX_THUNKS)
endif()
```